spec.html

```diff
@@ -24166,16 +24373,319 @@ <h1>Runtime Semantics: Evaluation</h1>
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-decoratorlistevaluation" type="sdo">
+      <h1>Runtime Semantics: DecoratorListEvaluation</h1>
+      <emu-grammar>DecoratorList : DecoratorList? Decorator</emu-grammar>
+      <emu-alg>
+        1. If |DecoratorList| is present, then
+          1. Let _leftValue_ be ? DecoratorListEvaluation of |DecoratorList|.
+        1. Else,
+          1. Let _leftValue_ be a new empty List.
+        1. Let _rightValue_ be ? DecoratorEvaluation of |Decorator|.
+        1. Prepend _rightValue_ to _leftValue_.
+        1. Return _leftValue_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-createdecoratoraccessobject" type="abstract operation">
+      <h1>
+        CreateDecoratorAccessObject (
+          _kind_: *"field"*, *"accessor"*, *"method"*, *"getter"*, or *"setter"*,
+          _name_: a Private Name,
+        )
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>Returns an object with appropriate accessor functions for th private value, for use in a decorator context object.</dd>
+      </dl>
+      <emu-alg>
+        1. Let _accessObj_ be ! OrdinaryObjectCreate(%Object.prototype%).
+        1. If _kind_ is *"field"*, *"accessor"*, *"method"*, or *"getter"*, then
+          1. Let _steps_ be the steps of a PrivateValueGetter function as specified below.
+          1. Let _length_ be the number of non-optional parameters of a PrivateValueGetter function as specified below.
+          1. Let _getter_ be ! CreateBuiltinFunction(_steps_, _length_, *""*, &laquo; [[PrivateStateKey]] &raquo;).
+          1. Set _getter_.[[PrivateStateKey]] to _name_.
+          1. Perform ! CreateDataPropertyOrThrow(_accessObj_, *"get"*, _getter_).
+        1. If _kind_ is *"field"*, *"accessor"*, or *"setter"*, then
+          1. Let _steps_ be the steps of a PrivateValueSetter function as specified below.
+          1. Let _length_ be the number of non-optional parameters of a PrivateValueSetter function as specified below.
+          1. Let _setter_ be ! CreateBuiltinFunction(_steps_, _length_, *""*, &laquo; [[PrivateStateKey]] &raquo;).
+          1. Set _setter_.[[PrivateStateKey]] to _name_.
+          1. Perform ! CreateDataPropertyOrThrow(_accessObj_, *"get"*, _setter_).
+        1. Return _accessObj_.
+      </emu-alg>
+      <p>A PrivateValueGetter function is an anonymous built-in function with a [[PrivateStateKey]] internal slot. When a PrivateValueGetter function that expects no arguments is called it performs the following steps:</p>
+      <emu-alg>
+        1. Let _o_ be the *this* value.
+        1. Let _f_ be the active function object.
+        1. Return ? PrivateGet(_f_.[[PrivateStateKey]], _o_).
+      </emu-alg>
+      <p>A PrivateValueSetter function is an anonymous built-in function with a [[PrivateStateKey]] internal slot. When a PrivateValueSetter function is called with argument _value_ it performs the following steps:</p>
```

Comment:
Can you use Abstract Closure to rewrite those built-in functions if possible? It will reduce the need for internal slots that are used to capture variables. An example: https://tc39.es/ecma262/multipage/control-abstraction-objects.html#sec-asyncgeneratorawaitreturn

```
1. Let _privateValueGetter_ be a new Abstract Closure with parameter (_value_) that captures _name_ and perform the following steps when called:
    1. Let _o_ be the *this* value.
    2. Perform ? PrivateSet(_name_, _o_, _value_).
    3. Return *undefined*
1. Let _f_ be ! CreateBuiltinFunction(_privateValueGetter_, 1, "", <<>>).
```